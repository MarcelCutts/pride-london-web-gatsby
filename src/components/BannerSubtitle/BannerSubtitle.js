import React from 'react'
import styled from 'styled-components'

const BannerSubtitle = styled.h3`
  color: #ffffff;
  font-size: 16px;

  & span {
    background-color: #2d2f7f;
    display: inline-block;
    padding: 0 10px;
  }
`
const StyledBannerSubtitle = ({ children }) => (
  <BannerSubtitle>
    <span>{children}</span>
  </BannerSubtitle>
)

export default StyledBannerSubtitle
