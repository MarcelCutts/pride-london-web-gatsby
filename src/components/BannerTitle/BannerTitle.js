import React from 'react'
import styled from 'styled-components'

const BannerTitle = styled.h1`
  display: block;
  color: #2d2f7f;
  font-size: 28px;
  margin-bottom: 0;

  & span {
    background-color: #ffffff;
    display: inline-block;
    padding: 0 10px;
  }
`

const StyledBannerTitle = ({ children }) => (
  <BannerTitle>
    <span>{children}</span>
  </BannerTitle>
)

export default StyledBannerTitle
