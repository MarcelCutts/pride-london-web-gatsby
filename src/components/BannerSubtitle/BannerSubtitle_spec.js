import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { shallow } from 'enzyme'

import BannerSubtitle from '../BannerSubtitle/BannerSubtitle'

describe('BannerSubtitle', () => {
  it('should render', () => {
    const tree = renderer.create(<BannerSubtitle />)
    expect(tree).toMatchSnapshot()
  })

  it('should render a <span />', () => {
    const wrapper = shallow(<BannerSubtitle />)
    expect(wrapper.find('span')).toHaveLength(1)
  })

  it('should render child text ', () => {
    const text = 'And here is a test subtitle!'
    const wrapper = shallow(<BannerSubtitle>{text}</BannerSubtitle>)
    expect(wrapper.find('span').text()).toBe(text)
  })
})
