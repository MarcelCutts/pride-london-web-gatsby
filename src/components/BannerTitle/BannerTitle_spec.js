import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { shallow } from 'enzyme'

import BannerTitle from '../BannerTitle/BannerTitle'

describe('BannerTitle', () => {
  it('should render', () => {
    const tree = renderer.create(<BannerTitle />)
    expect(tree).toMatchSnapshot()
  })

  it('should render a <span />', () => {
    const wrapper = shallow(<BannerTitle />)
    expect(wrapper.find('span')).toHaveLength(1)
  })

  it('should render child text ', () => {
    const text = 'Here is a test title!'
    const wrapper = shallow(<BannerTitle>{text}</BannerTitle>)
    expect(wrapper.find('span').text()).toBe(text)
  })
})
