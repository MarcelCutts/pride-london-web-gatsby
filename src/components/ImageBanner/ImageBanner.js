import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import BannerTitle from '../BannerTitle/BannerTitle'
import BannerSubtitle from '../BannerSubtitle/BannerSubtitle'

const Container = styled.div`
  display: flex;
  flex-direction: column;
  height: 210px;
  justify-content: center;
  line-height: 0;
  padding-left: 10px;

  @media (min-width: 768px) {
    height: 250px;
    padding-left: 90px;
  }
`

const ImageBanner = ({ titleText, subtitleText, imageSrc, altText }) => (
  <Container>
    <img src={imageSrc} alt={altText} />
    <BannerTitle>{titleText}</BannerTitle>
    <BannerSubtitle>{subtitleText}</BannerSubtitle>
  </Container>
)

ImageBanner.propTypes = {
  imageSrc: PropTypes.string,
  altText: PropTypes.string,
  subtitleText: PropTypes.string,
  titleText: PropTypes.string,
}

ImageBanner.defaultProps = {
  imageSrc: '',
  altText: '',
  subtitleText: '',
  titleText: '',
}

export default ImageBanner
