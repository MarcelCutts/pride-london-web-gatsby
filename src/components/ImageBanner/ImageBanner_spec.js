import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { shallow, mount } from 'enzyme'

import ImageBanner, { BackgroundImage } from './ImageBanner'
import BannerTitle from '../BannerTitle/BannerTitle'
import BannerSubtitle from '../BannerSubtitle/BannerSubtitle'

describe('ImageBanner', () => {
  it('should render', () => {
    const wrapper = shallow(<ImageBanner />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render a <BannerTitle />', () => {
    const wrapper = shallow(<ImageBanner />)
    expect(wrapper.find(BannerTitle)).toHaveLength(1)
  })

  it('should render a <BannerSubtitle />', () => {
    const wrapper = shallow(<ImageBanner />)
    expect(wrapper.find(BannerSubtitle)).toHaveLength(1)
  })

  it('should render the titleText from props to BannerTitle ', () => {
    const titleText = 'Here is a test title!'
    const wrapper = mount(<ImageBanner titleText={titleText} />)
    expect(wrapper.find(BannerTitle).text()).toBe(titleText)
  })

  it('should render the subtitleText from props to BannerTitle ', () => {
    const subtitleText = 'And here is a test subtitle!'
    const wrapper = mount(<ImageBanner subtitleText={subtitleText} />)
    expect(wrapper.find(BannerSubtitle).text()).toBe(subtitleText)
  })

  it('should pass the imageSrc prop to the BackgroundImage component', () => {
    const imageSrc = 'http://www.pics.com/coolpic'
    const wrapper = shallow(<ImageBanner imageSrc={imageSrc} />)
    expect(wrapper.find(BackgroundImage).props().imageSrc).toBe(imageSrc)
  })
})
