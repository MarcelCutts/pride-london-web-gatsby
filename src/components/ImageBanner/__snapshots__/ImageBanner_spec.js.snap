// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageBanner should render 1`] = `
.c1 {
  display: block;
  color: #2d2f7f;
  font-size: 28px;
  margin-bottom: 0;
}

.c1 span {
  background-color: #ffffff;
  display: inline-block;
  padding: 0 10px;
}

.c2 {
  color: #ffffff;
  font-size: 16px;
}

.c2 span {
  background-color: #2d2f7f;
  display: inline-block;
  padding: 0 10px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 210px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 0;
  padding-left: 10px;
}

@media (min-width:768px) {
  .c0 {
    height: 250px;
    padding-left: 90px;
  }
}

<div
  className="c0"
>
  <img
    alt=""
    src=""
  />
  <h1
    className="c1"
  >
    <span>
      
    </span>
  </h1>
  <h3
    className="c2"
  >
    <span>
      
    </span>
  </h3>
</div>
`;
